diff --git a/nemo_rl/algorithms/sft.py b/nemo_rl/algorithms/sft.py
index a447510..e987002 100644
--- a/nemo_rl/algorithms/sft.py
+++ b/nemo_rl/algorithms/sft.py
@@ -138,6 +138,7 @@ def setup(
         shuffle=data_config["shuffle"],
         collate_fn=rl_collate_fn,
         drop_last=True,
+        num_workers=10,
     )

     if last_checkpoint_path is not None:
@@ -152,6 +153,7 @@ def setup(
         shuffle=False,
         collate_fn=rl_collate_fn,
         drop_last=False,
+        num_workers=10,
     )

     # ==========================
